src/redux/books/book.js

```diff
@@ -50,8 +29,45 @@ export default function bookReducer(state = defaultState, action) {
     case REMOVE_BOOK:
       return [
         ...state.filter((book) => book.id !== action.id)];
-
+    case GET_BOOK:
+      return [...action.payload];
     default:
       return state;
   }
 }
+
+export const serverBooks = () => async (dispatch) => {
+  const response = await fetch(baseUrl);
+  const data = await response.json();
+  const books = [];
+  Object.keys(data).forEach((e) => {
+    books.push({ ...data[e][0], item_id: e });
+  });
+  dispatch({
+    type: GET_BOOK,
+    payload: books,
+  });
+};
+export const addBooks = (book) => async (dispatch) => {
+  await fetch(baseUrl, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(book),
+  });
+  dispatch({
+    type: ADD_BOOK,
+    payload: book,
+  });
+};
+
+export const removeBooks = (id) => async (dispatch) => {
+  await fetch(`${baseUrl}/${id}`, {
+    method: 'DELETE',
+  });
+  dispatch({
+    type: REMOVE_BOOK,
+    payload: id,
+  });
+};
```

Comment:
Also, removing a book removes all books from the state.

![image](https://user-images.githubusercontent.com/38088818/179000171-da1cc041-3880-4c9a-9c04-feafa15666c2.png)

- Kindly fix your redux actions to make sure that only the selected book is removed at a time.

[OPTIONAL] To be able to see actions triggered with the previous and current state as demonstrated above, you can install `redux-logger` and apply it as middleware in your `configureStore.js` file.

![image](https://user-images.githubusercontent.com/38088818/179001516-d8800fad-5b88-4260-a073-b41f1eefca33.png)